### PR TITLE
Update rightLabel when clicked while player paused

### PR DESCRIPTION
**Pull request by decodism**
- [ ] This change has been discussed with the author.
- [x] It implements / fixes issue #3849.

### CODE REVIEW

File: iina/MPVController.swift [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ class MPVController: NSObject {
             // video is paused to avoid wasting energy with needless processing. If paused shutdown
             // the timer that synchronizes the UI and the high priority display link thread.
             if paused {
-              player.invalidateTimer()
+              if ![.willShow, .shown].contains(player.mainWindow.animationState) { player.invalidateTimer() }
```
Comment (low-batt) [inline iina/MPVController.swift:1028]: This fixes the problem, but is undesirable from an energy use perspective. We absolutely do not want the timer to run when playback is paused and therefore the video position is not changing.

**Review by low-batt — CHANGES_REQUESTED** [review]
The timer that updates the UI is still running when it should not, wasting energy. That is why this problem seemed to be fixed.  The fix for this issue should not rely upon that timer running. We want to get to the point where that timer is only run when playback is active and the OSC is visible. We have been moving in that direction, but more changes are needed. So this fix can not depend up the timer updating the label.

I suggest the fix follow the normal pattern for handling changes in a control. 

One way to do this would be to add a `didSet` property observer to the `DurationDisplayTextField` `mode` property. This observer would call `updateText` to update the `stringValue` `NSTextField` property. Then the observer would set [needsDisplay](https://developer.apple.com/documentation/appkit/nsview/1483360-needsdisplay) to true.

To be able to call `updateText` `DurationDisplayTextField` needs to know the duration and position. That means adding properties and changing `updateText` to remember the values used to generate the string.

Something like that.

**Comment by decodism** [comment]
I reverted to my previous fix and implemented your suggestion.

**Review by low-batt — APPROVED** [review]
Changes look good. I pulled the PR locally and confirmed the fix works.

**Review by uiryuu — APPROVED** [review]

**Comment by uiryuu** [comment]
Thanks for your PR!

**Pull request merged**